### PR TITLE
[RVC2] getStereoPairs

### PR DESCRIPTION
**Pull request by zrezke**
Returns stereo pairs on known devices. Empty on FFC, because the stereo pairs could be anything. Skip getAvailableStereoPairs for now.

### CODE REVIEW

File: src/device/DeviceBase.cpp [hunk 1]
```diff
@@ -1049,6 +1049,94 @@ std::vector<CameraBoardSocket> DeviceBase::getConnectedCameras() {
     return pimpl->rpcClient->call("getConnectedCameras").as<std::vector<CameraBoardSocket>>();
 }
 
+std::vector<StereoPair> DeviceBase::getAvailableStereoPairs() {
+    std::vector<dai::StereoPair> stereoPairs;
+    std::function<std::vector<dai::CameraBoardSocket>(dai::CameraBoardSocket, const std::unordered_map<dai::CameraBoardSocket, dai::CameraInfo>&, std::vector<dai::CameraBoardSocket>&)> visitLinks;
+    visitLinks = [&visitLinks](dai::CameraBoardSocket camId, const std::unordered_map<dai::CameraBoardSocket, dai::CameraInfo>& camData, std::vector<dai::CameraBoardSocket>& visited) -> std::vector<dai::CameraBoardSocket> {
```
Comment (themarpe) [inline src/device/DeviceBase.cpp:1055]: Perhaps the following would be simpler (if it goes through)
```suggestion
    auto visitLinks = [&visitLinks](dai::CameraBoardSocket camId, const std::unordered_map<dai::CameraBoardSocket, dai::CameraInfo>& camData, std::vector<dai::CameraBoardSocket>& visited) -> std::vector<dai::CameraBoardSocket> {
```

Comment (zrezke) [reply src/device/DeviceBase.cpp:1055]: Unfortunately can't use auto here..
```
error: use of ‘visitLinks’ before deduction of ‘auto’
```

Comment (themarpe) [reply src/device/DeviceBase.cpp:1055]: Oh right, recursion

File: src/device/DeviceBase.cpp [hunk 2]
```diff
@@ -1049,6 +1049,94 @@ std::vector<CameraBoardSocket> DeviceBase::getConnectedCameras() {
...
+    }
+    // Step 1: Find the linked cameras
+    std::unordered_map<dai::CameraBoardSocket, std::vector<dai::CameraBoardSocket>> linkedCameras;
+    for (auto const& [cam, camInfo] : eepromData.cameraData) {
+        auto camId = cam;
+        auto extrinsics = camInfo.extrinsics;
+        auto toSocket = extrinsics.toCameraSocket;
+        std::vector<dai::CameraBoardSocket> visited;
+        linkedCameras[camId] = visitLinks(toSocket, eepromData.cameraData, visited);
+    }
```
[the remark this answers is not in the document]
Comment (zrezke) [reply src/device/DeviceBase.cpp:1089]: Good catch, we can.

[the remark this answers is not in the document]
Comment (zrezke) [reply src/device/DeviceBase.cpp]: Addressed in https://github.com/luxonis/depthai-core/pull/941/commits/88d383f151fc14457c4537bac5932daf66119fd7

**Pull request merged**